R/lsmeans.R

```diff
@@ -34,10 +44,42 @@ lscombinations <- function(data, mterms, fix) {
 }
 
 
-#' Title - TODO
-#' 
-#' @param model TODO
-#' @param ... TODO
+#' Least Square Means
+#'
+#' Estimates the least square means from a linear model. This is
+#' essentially where we generate a prediction from the model
+#' by fitting it to some hypothetical observation that is constructed
+#' by averaging the data. See details for more information.
+#'
+#' @details
+#' Numeric variables are evaluated at the mean across the entire dataset
+#' (after removing missing values)
+#' Factor variables are evaluated at all levels (including combinations
+#' with other factor variables) with the final return value being
+#' average across all the predictions generated at each of these levels.
+#'
+#' Use the `...` argument to fix specific variables to specific values.
+#'
+#' See the references for identical implementations as done in SAS and via
+#' the emmeans package. This function attempts to re-implement the
```

Comment:
Maybe it is worth mentioning explicitly which function of emmeans we are referring of?

Reply:
Unfortunately its not one function but several :(